app/services/concerns/xccdf/rule_group_relationships.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Xccdf
+  # Methods related to saving RuleGroupRelationships
+  module RuleGroupRelationships
+    extend ActiveSupport::Concern
+    included do
+      def save_rule_group_relationships
+        @op_rules_and_rule_groups = @op_rule_groups + @op_rules
+        @rule_group_relationships = all_rule_group_relationships
+
+        ::RuleGroupRelationship.import!(
+          @rule_group_relationships.select(&:new_record?), ignore: true
+        )
+
+        rgr_base = ::RuleGroupRelationship.left_outer_joins(dependent_rule_group: :benchmark)
+        rgr_links_to_remove(rgr_base).delete_all
+      end
+
+      private
+
+      def all_rule_group_relationships
+        @all_rule_group_relationships ||= @op_rules_and_rule_groups.flat_map do |op_r_or_rg|
+          with_conflicts(op_r_or_rg) + with_requires(op_r_or_rg)
+        end.compact
+      end
+
+      def with_conflicts(op_r_or_rg)
+        left = rule_or_rule_group_for(ref_id: op_r_or_rg.id)
+        op_r_or_rg.conflicts.map do |conflicting_ref_id|
+          right = rule_or_rule_group_for(ref_id: conflicting_ref_id)
+          next unless right
+
+          ::RuleGroupRelationship.find_or_initialize_by(
+            left: left, right: right, relationship: 'conflicts'
+          )
+        end
+      end
+
+      def with_requires(op_r_or_rg)
+        left = rule_or_rule_group_for(ref_id: op_r_or_rg.id)
+        op_r_or_rg.requires.map do |required_ref_id|
+          right = rule_or_rule_group_for(ref_id: required_ref_id)
+          next unless right
+
+          ::RuleGroupRelationship.find_or_initialize_by(
+            left: left, right: right, relationship: 'requires'
+          )
+        end
+      end
+
+      def rgr_links_to_remove(base)
+        grouped_by_rgr = @rule_group_relationships&.group_by(&:left_id)
+        grouped_by_rgr&.reduce(RuleGroupRelationship.none) do |query, (left_id, rgr)|
+          query.or(
+            base.where(left_id: left_id)
+                .where.not(relationship: rgr.map(&:relationship))
+                .where.not(right_id: rgr.map(&:right_id))
+          )
+        end
+      end
```

Comment:
@skateman I don't think this method is set up correctly because I'm kind of confused by what the purpose of it is? I thought that I understood what the similar method in ProfileRules was doing but I'm not sure if I do.. When should we be removing the links?

Reply:
When something like this is too crazy to read, you can always put call `to_sql` on it and read the query :wink: